samza-core/src/test/java/org/apache/samza/util/TestConfigUtil.java

```diff
@@ -161,6 +162,31 @@ public void testApplyRewriterClassDoesNotExist() {
     assertEquals(expectedConfig, ConfigUtil.applyRewriter(new MapConfig(fullConfig), REWRITER_NAME));
   }
 
+  @Test
+  public void testLoadConfigWithoutLoader() {
+    Map<String, String> config = new HashMap<>();
+    config.put(JobConfig.JOB_NAME, "new-test-job");
+
+    Config actual = ConfigUtil.loadConfig(new MapConfig(config));
+
+    assertEquals(config.size(), actual.size());
+    assertEquals("new-test-job", actual.get(JobConfig.JOB_NAME));
+  }
+
+  @Test
+  public void testLoadConfigWithLoader() {
```

Comment:
Should this also test config rewriting and overriding?

Reply:
Updated to include both overrides and rewrites.